java-checks/src/main/java/org/sonar/java/checks/unused/UnusedTestRuleCheck.java

```diff
@@ -47,11 +48,16 @@ public List<Tree.Kind> nodesToVisit() {
   @Override
   public void visitNode(Tree tree) {
     ClassTree classTree = (ClassTree) tree;
+    boolean isAbstract = classTree.modifiers().modifiers().stream().anyMatch(m -> Modifier.ABSTRACT == m.modifier());
     for (Tree member : classTree.members()) {
       if (member.is(Tree.Kind.VARIABLE)) {
         VariableTree variableTree = (VariableTree) member;
         Symbol symbol = variableTree.symbol();
         if ((isTestNameOrTemporaryFolderRule(symbol) || hasTempDirAnnotation(symbol)) && symbol.usages().isEmpty()) {
+          // if class is abstract, then we need to check modifier - if not private, then it's okay
+          if (isAbstract && variableTree.modifiers().modifiers().stream().noneMatch(m -> Modifier.PRIVATE == m.modifier())) {
```

Comment:
The ` variableTree.modifiers().modifiers().stream().noneMatch(m -> Modifier.PRIVATE == m.modifier())` can be simplified using ModifierUtils: `!ModifiersUtils.hasModifier(variableTree.modifiers(), Modifier.PRIVATE)`